sdks/java/io/xml/src/test/java/org/apache/beam/sdk/io/xml/XmlIOIT.java

```diff
@@ -0,0 +1,183 @@
+ * <pre>
+ *  mvn -e -Pio-it verify -pl sdks/java/io/xml
+ *  -Dit.test=org.apache.beam.sdk.io.xml.XmlIOIT
+ *  -DintegrationTestPipelineOptions='[
+ *  "--numberOfRecords=100000",
+ *  "--filenamePrefix=output_file_path",
+ *  "--charset=UTF-8",
+ *  ]'
+ * </pre>
+ * </p>
+ */
+@RunWith(JUnit4.class)
+public class XmlIOIT {
+
+  private static final Map<Integer, String> EXPECTED_HASHES = ImmutableMap.of(
+    1000, "7f51adaf701441ee83459a3f705c1b86",
+    100_000, "af7775de90d0b0c8bbc36273fbca26fe",
+    100_000_000, "bfee52b33aa1552b9c1bfa8bcc41ae80"
+  );
+
+  private static Integer numberOfRecords;
+
+  private static String filenamePrefix;
+
+  private static Charset charset;
+
+  @Rule
+  public TestPipeline pipeline = TestPipeline.create();
+
+  @Rule
+  public TestPipeline readPipeline = TestPipeline.create();
+
+  @BeforeClass
+  public static void setup() {
+    PipelineOptionsFactory.register(IOTestPipelineOptions.class);
+    IOTestPipelineOptions options = TestPipeline
+      .testingPipelineOptions()
+      .as(IOTestPipelineOptions.class);
+
+    filenamePrefix = appendTimestampSuffix(options.getFilenamePrefix());
+    numberOfRecords = options.getNumberOfRecords();
+    charset = Charset.forName(options.getCharset());
+  }
+
+  @Test
+  public void writeThenReadViaSinkAndReadFiles() {
+    PCollection<String> testFileNames = pipeline
+      .apply("Generate sequence", GenerateSequence.from(0).to(numberOfRecords))
+      .apply("Create Birds", MapElements.via(new LongToBird()))
+      .apply("Write birds to xml files", FileIO.<Bird>write()
+          .via(XmlIO.sink(Bird.class)
+            .withRootElement("birds")
+            .withCharset(charset))
+          .to(filenamePrefix)
+          .withPrefix("birds")
+          .withSuffix(".xml"))
+      .getPerDestinationOutputFilenames()
+      .apply("Prevent fusion", Reshuffle.viaRandomKey())
+      .apply("Get file names", Values.create());
+
+    PCollection<Bird> birds = testFileNames
+      .apply("Find files", FileIO.matchAll())
+      .apply("Read matched files", FileIO.readMatches())
+      .apply("Read xml files", XmlIO.<Bird>readFiles()
+        .withRecordClass(Bird.class).withRootElement("birds")
+        .withRecordElement("bird")
+        .withCharset(charset));
+
+    PCollection<String> consolidatedHashcode = birds
+      .apply("Map birds to strings", MapElements.via(new BirdToString()))
+      .apply("Calculate hashcode", Combine.globally(new HashingFn()));
+
+    String expectedHash = IOTestHelper.getHashForRecordCount(numberOfRecords, EXPECTED_HASHES);
+    PAssert.thatSingleton(consolidatedHashcode).isEqualTo(expectedHash);
+
+    testFileNames.apply("Delete test files", ParDo.of(new DeleteFileFn())
+        .withSideInputs(consolidatedHashcode.apply(View.asSingleton())));
+
+    pipeline.run().waitUntilFinish();
+  }
+
+  @Test
+  public void writeThenReadViaWriteAndRead() {
```

Comment:
Not sure what this name means. Could you please update.

Reply:
Ok